src/ssh-compute.ts

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import uuid from 'uuid';
+import os from 'os';
+
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as toolCache from '@actions/tool-cache';
+import * as setupGcloud from '@google-github-actions/setup-cloud-sdk';
+import {
+  exactlyOneOf,
+  isPinnedToHead,
+  pinnedToHeadWarning,
+} from '@google-github-actions/actions-utils';
+
+import path from 'path';
+import { promises as fs } from 'fs';
+
+import {
+  SSH_PUBLIC_KEY_FILENAME,
+  SSH_PRIVATE_KEY_FILENAME,
+  GOOGLE_SSH_KEYS_TEMP_DIR_VAR,
+} from './const';
+
+export const GCLOUD_METRICS_ENV_VAR = 'CLOUDSDK_METRICS_ENVIRONMENT';
+export const GCLOUD_METRICS_LABEL = 'github-actions-ssh-compute';
+
+export function parseFlags(flags: string): RegExpMatchArray {
+  return flags.match(/(".*?"|[^"\s=]+)+(?=\s*|\s*$)/g)!; // Split on space or "=" if not in quotes
+}
+
+/**
+ * Executes the main action. It includes the main business logic and is the
+ * primary entry point. It is documented inline.
+ */
+export async function run(): Promise<void> {
+  core.exportVariable(GCLOUD_METRICS_ENV_VAR, GCLOUD_METRICS_LABEL);
+
+  // Warn if pinned to HEAD
+  if (isPinnedToHead()) {
+    core.warning(pinnedToHeadWarning('v0'));
+  }
+
+  // Get inputs
+  // Core inputs
+  let instanceName = core.getInput('instance_name');
+  const zone = core.getInput('zone');
+  const user = core.getInput('user');
+  const ssh_public_key = core.getInput('ssh_public_key');
+  const ssh_private_key = core.getInput('ssh_private_key');
+  const ssh_keys_dir = core.getInput('ssh_keys_dir') || `${os.tmpdir()}/${uuid.v4()}`;
```

Comment:
We shouldn't use UUIDs for anything, especially directory paths. There are helpers in `actions-utils` for creating random (and secure) directories.